gitea/giteaclient.go

```diff
@@ -0,0 +1,404 @@
+/*
+Copyright 2023 The Flux CD contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gitea
+
+import (
+	"context"
+	"fmt"
+
+	"code.gitea.io/sdk/gitea"
+	"github.com/fluxcd/go-git-providers/gitprovider"
+)
+
+// giteaClientImpl is a wrapper around *gitea.Client, which implements higher-level methods,
+// operating on the gitea structs. TODO: Verify pagination is implemented for all List* methods,
+// all returned objects are validated, and HTTP errors are handled/wrapped using handleHTTPError.
+// This interface is also fakeable, in order to unit-test the client.
+type giteaClient interface {
```

Comment:
I would understand the need for another interface here if we had more than one implementation of it but we don't. The only implementation this PR ships is `giteaClientImpl`. Let's just get rid of this interface until we really need it.